client/src/component/Header.tsx

```diff
@@ -0,0 +1,52 @@
+import { useNavigate } from "react-router-dom";
+import { useLogin } from "../store/LoginState";
+
+export default function Header() {
+  // 로그인 여부에 따라 다른 헤더 출력
+  const navigate = useNavigate();
+  const { isLogin, setIsLogin } = useLogin();
+
+  const goToMain = () => {
+    navigate("/");
+  };
+
+  const goToHeadline = () => {
+    navigate("/headline");
+  };
+
+  const goToFavorites = () => {
+    navigate("/");
+  };
+
+  const handleClick = () => {
+    if (!isLogin) {
+      navigate("/login");
+    }
+    if (isLogin) {
+      alert("로그아웃 완료");
+      setIsLogin(false);
+      navigate("/");
+    }
+  };
```

Comment:
isLogin 에 따라서 effect 분리되는 것 같아요.

```suggestion
  const handleClick = () => {
    if (!isLogin) {
      navigate("/login");
      return;
    } else {
      alert("로그아웃 완료");
      setIsLogin(false);
      navigate("/");
      return;
    }
  };
```

요렇게 변경되면 더 낫지 않을까요??
https://jheloper.github.io/2019/06/write-early-return-code/
요글 한번 읽어보시면 좋을 것 같아요!